src/engine/archive.cpp

```diff
@@ -0,0 +1,125 @@
+#include "engine/archive.h"
+#include "engine/engine.h"
+#include "flow/flow.h"
+#include "roadnet/roadnet.h"
+#include "roadnet/trafficlight.h"
+#include "vehicle/vehicle.h"
+
+namespace CityFlow {
+
+    Archive::Archive(const Engine &engine)
+    : step(engine.step), activeVehicleCount(engine.activeVehicleCount), rnd(engine.rnd) {
+        // copy the vehicle Pool
+
+        for (const auto &veh : engine.vehiclePool) {
+            const Vehicle *oldVehicle = veh.second.first;
+            Vehicle * newVehicle = new Vehicle(*oldVehicle);
+            vehiclePool.emplace(oldVehicle->getPriority(), std::make_pair(newVehicle, veh.second.second));
```

Comment:
Does copying during construction mean one archive can be loaded only once?

Reply:
> Does copying during construction mean one archive can be loaded only once?

This is a mistake caused by my neglect. I think there are two choices:

-  Set a flag in each Archive object, which stands for whether this archive has been 'used'. Thus we can avoid twice loading.  The user need to create another copy manually.

- After an archive is loaded, construct a new archive immediately and replace the former one.

The first way is better in performance but easy to cause mistakes while the latter is easy to use but may cause waste.